libs/chain-signatures/contract/src/lib.rs

```diff
@@ -287,6 +323,19 @@ impl VersionedMpcContract {
                         Err(InvalidParameters::RequestNotFound.into())
                     }
                 }
+                Self::V1(mpc_contract) => {
```

Comment:
It seems all of this logic is identical to V0 above. Maybe you could add a `VersionedMpcContract::get_pending_request(...) -> Option<YieldIndex>` ([like this pattern used in nearcore](https://github.com/near/nearcore/blob/8097202bfa56dba789a4f7fcdb680c2cdaf258b2/core/primitives/src/transaction.rs#L81-L87)) and avoid duplicating the rest of the code.